lib/lf/geometry/test/geometry_tests.cc

```diff
@@ -1,559 +1,301 @@
-        Eigen::MatrixXd alphaBeta = paddedSubNodeCoords.transpose()
-                                        .fullPivLu()
-                                        .solve(mapNodeCoords.transpose())
-                                        .transpose();
-
-        // map points onto RefEl
-        Eigen::MatrixXd paddedPoints =
-            Eigen::MatrixXd::Ones(points.rows() + 1, points.cols());
-        paddedPoints.block(0, 0, points.rows(), points.cols()) = points;
-        const Eigen::MatrixXd mappedPoints = alphaBeta * paddedPoints;
-
-        // map coordinates in subRefEl.Dimension to geom.DimGlobal
-        auto globalPointsFromSub = subGeom->Global(points);
-        // map coordinates in RefEl.Dimension to geom.DimGlobal
-        auto globalPoints = geom.Global(mappedPoints);
-
-        EXPECT_TRUE(globalPoints.isApprox(globalPointsFromSub))
-            << "Global mapping of points " << points << " from subEntity "
-            << subEntity << " in relative codim " << codim
-            << " differs from global mapping";
-      }
-    }
-  }
-}
+namespace lf::geometry::test {
 
 void runGeometryChecks(const lf::geometry::Geometry &geom,
                        const Eigen::MatrixXd &eval_points,
                        const double tolerance) {
-  checkJacobian(geom, eval_points, tolerance);
+  lf::geometry::test_utils::checkJacobian(geom, eval_points, tolerance);
   // JacobianInverseGramian is not defined for points
   if (geom.RefEl() != lf::base::RefEl::kPoint()) {
-    checkJacobianInverseGramian(geom, eval_points);
+    lf::geometry::test_utils::checkJacobianInverseGramian(geom, eval_points);
   }
-  checkIntegrationElement(geom, eval_points);
-  checkSubGeometry(geom, [](lf::base::RefEl refEl) -> lf::quad::QuadRule {
-    return lf::quad::make_QuadRule(refEl, 5);
-  });
+  lf::geometry::test_utils::checkIntegrationElement(geom, eval_points);
+  lf::geometry::test_utils::checkSubGeometry(
+      geom, [](lf::base::RefEl refEl) -> lf::quad::QuadRule {
+        return lf::quad::make_QuadRule(refEl, 5);
+      });
 }
 
-/**
- * Check if the total volume is conserved after call to ChildGeometry()
- */
-void checkChildGeometryVolume(
-    const lf::geometry::Geometry &geom, const lf::refinement::RefPat &refPat,
-    const lf::base::sub_idx_t &anchor = lf::refinement::idx_nil) {
-  // compute volume by means of overkill quadrature
-  auto computeVolume = [](const lf::geometry::Geometry &geom) {
-    const auto qr = lf::quad::make_QuadRule(geom.RefEl(), 23);
-
-    const auto &points = qr.Points();
-    const auto &weights = qr.Weights();
-    const auto &integrationElements = geom.IntegrationElement(points);
-
-    double vol = 0.;
-
-    for (size_t j = 0; j < points.cols(); ++j) {
-      vol += weights(j) * integrationElements(j);
-    }
-
-    return vol;
-  };
+TEST(Geometry, Point) {
+  Eigen::MatrixXd global_nodes_2D(2, 1);
+  global_nodes_2D << 1, 2;
+  Eigen::MatrixXd global_nodes_3D(3, 1);
+  global_nodes_3D << 3, 4, 0;
```

Comment:
And here you would write `Eigen::Vector3d global_nodes_3D(3,4,0)`

Reply:
Unfortunately, this doesn't work as the initializer list of the for-loop complains about conflicting types if I mix `Vector2d` and `Vector3d`.